src/browser/web_socket.rs

```diff
@@ -131,7 +132,8 @@ impl WebSocket {
     ///
     /// Returns error when JSON serialization or sending fails.
     pub fn send_json<T: Serialize + ?Sized>(&self, data: &T) -> Result<()> {
-        let data = swb::to_value(data)?
+        let data = JSON::stringify(&swb::to_value(data)?)
+            .map_err(|_| WebSocketError::ConversionError)?
             .as_string()
             .ok_or(WebSocketError::ConversionError)?;
```

Comment:
Since `JSON::stringify` returns a `JsString`, you can instead use infallible `String::from(...)` on it (or `.into()` equivalent).

Reply:
That's way better, thank you. I must have somehow missed that those trait instances existed.

Reply:
```rust
impl<'a> From<&'a JsString> for String {
    fn from(s: &'a JsString) -> Self {
        s.obj.as_string().unwrap_throw()
    }
}
```
I thought `From` was never supposed to panic.

Reply:
It won't because type is statically known here.

Reply:
Also, `.unwrap_throw()` doesn't panic in Rust terms, it's much more lightweight method as it immediately throws JS exception that unwinds entire JS stack.